pygsti/forwardsims/matrixforwardsim.py

```diff
@@ -1243,7 +1249,20 @@ def _dprobs_from_rho_e(self, spam_tuple, rho, e, gs, d_gs, scale_vals, wrt_slice
         # dp_dOps[i,j] = dot( e, dot( d_gs, rho ) )[0,i,j,0]
         # dp_dOps      = squeeze( dot( e, dot( d_gs, rho ) ), axis=(0,3))
         old_err2 = _np.seterr(invalid='ignore', over='ignore')
-        dp_dOps = _np.squeeze(_np.dot(e, _np.dot(d_gs, rho)), axis=(0, 3)) * scale_vals[:, None]
+        #print(f'{d_gs.shape=}')
+        #print(f'{e.shape=}')
+        #print(f'{rho.shape=}')
+        #print(f'{_np.dot(d_gs, rho).shape=}')
+        #print(f'{_np.dot(e, _np.dot(d_gs, rho)).shape=}')
+        #print(f'{_np.squeeze(_np.dot(e, _np.dot(d_gs, rho)), axis=(0, 3)).shape=}')
+        #
+        #print(f"{_np.einsum('hk,ijkl,lm->ij', e, d_gs, rho).shape=}")
+        #
+        #print(f"{_np.linalg.norm(_np.squeeze(_np.dot(e, _np.dot(d_gs, rho))) - _np.einsum('hk,ijkl,lm->ij', e, d_gs, rho))=}")
```

Comment:
I'd prefer we not have long blocks of commented-out debugging code. If you wrap this with triple-quotes as an in-line comment then we can at least collapse it in an IDE.

Reply:
Thanks for catching this, it has been removed.